File: swift/day6_part1_2020.swift

import Foundation

if let input = try? String(contentsOfFile: "input.txt") {
    let groups = input.components(separatedBy: "\n\n")
    let sum = groups.reduce(0) { result, group in
        let answers = group.replacingOccurrences(of: "\n", with: "")
        let uniqueAnswers = Set(answers)
        return result + uniqueAnswers.count
    }
    print(sum)
}

File: training_data/reflection-tuning/day6_part1_2020.clj
(ns day6.core
  (:require [clojure.string :as str]))

(defn count-yes-answers [group]
  (->> group
       (mapcat seq)
       set
       count))

(defn solve-custom-customs [input]
  (->> (str/split input #"\n\n")
       (map #(str/split % #"\n"))
       (map count-yes-answers)
       (reduce +)))

(defn -main []
  (let [input (slurp "input.txt")
        result (solve-custom-customs input)]
    (println result)))

(-main)
